modules/linear_boltzmann_solvers/lbs_solver/iterative_methods/ags_linear_solver.h

```diff
@@ -12,55 +12,38 @@ namespace lbs
 {
 
 /**Linear Solver specialization for Within GroupSet (WGS) solves.*/
-class AGSLinearSolver : public LinearSolver
+class AGSLinearSolver
 {
+private:
+  std::shared_ptr<LinearSolverContext> context_ptr_;
```

Comment:
This would be a great opportunity to get rid of an unnecessary context. There is no reason we need a context to store a list of `WGSSolvers`. This can be stored in this class.

Reply:
Good point. That's all it does here. Let me see about removing it completely.

Reply:
I've removed AGSContext.